src/main/kotlin/calculator/Operator.kt

```diff
@@ -0,0 +1,21 @@
+package calculator
+
+enum class Operator(
+    private val sign: String,
+    private val calculation: (Number, Number) -> Number
+) {
+    ADDITION("+", { a, b -> Number(a.value + b.value) }),
+    SUBTRACTION("-", { a, b -> Number(a.value - b.value) }),
+    MULTIPLICATION("*", { a, b -> Number(a.value * b.value) }),
+    DIVISION("/", { a, b -> Number(a.value / b.value) }),
+    ;
+
+    fun calculate(a: Number, b: Number): Number = calculation(a, b)
+
+    companion object {
+        fun findBySign(sign: String): Operator {
+            return values().firstOrNull { it.sign == sign }
+                ?: throw IllegalArgumentException("$sign 에 해당하는 연산자를 찾지 못했습니다.")
+        }
```

Comment:
firstOrNull 보다는, find 확장 함수가 더 의미에 와닿아 보입니다
```suggestion
        fun findBySign(sign: String): Operator {
            return values().find { it.sign == sign }
                ?: throw IllegalArgumentException("$sign 에 해당하는 연산자를 찾지 못했습니다.")
        }
```

Reply:
!!! 매우 동의합니다! `find` 내부 설명을 찾아보니 정확히 원하던 동작을 하네요.